pyfixest/estimation/feiv_.py

```diff
@@ -193,3 +206,225 @@ def get_fit(self) -> None:
+
+            # Use fitted values from the first stage
+            self._X_hat = model1._Y_hat_link
+
+            # Residuals from the first stage
+            self._v_hat = model1._u_hat
+
+            # Store 1st stage model for further use
+            self._model_1st_stage = model1
+        else:
+            raise TypeError("The first stage model must be of type Feols")
+
+    def IV_Diag(self) -> None:
+        """Implement IV weakness test(F-test)."""
+        # This method covers hetero-robust and clustered-robust F stats
+        # This method produces two statistics
+        #   self._f_stat_1st_stage : F statistics of first stage regression
+        #   self._eff_F : Effective F statistics (Olea and Pflueger 2013)
+        #                 of first stage regression
+        #
+        # The following is the usage example of this method
+        # Note that "self._f_stat_1st_stage" is adjusted to the specification of
+        # vcov. If vcov_detail = "iid", F statistics is not adjusted, otherwise
+        # it is always adjusted.
+        ###################################################################
+        """
+        #import numpy as np
+        #import pandas as pd
+
+        #from pyfixest.estimation.estimation import feols
+
+        ##Set random seed for reproducibility
+        #np.random.seed(1)
+
+        ## Number of observations
+        #n = 1000
+
+        ## Simulate the data
+        ## Instrumental variable
+        #z = np.random.binomial(1, 0.5, size=n)
+        #z2 = np.random.binomial(1, 0.5, size=n)
+        # Endogenous variable
+        #d = 0.5 * z + 1.5 * z2 + np.random.normal(size=n)
+
+        ## Control variables
+        #c1 = np.random.normal(size=n)
+        #c2 = np.random.normal(size=n)
+
+        ## Outcome variable
+        #y = 1.0 + 1.5 * d + 0.8 * c1 + 0.5 * c2 + np.random.normal(size=n)
+
+        ## Cluster variable
+        #cluster = np.random.randint(1, 50, size=n)
+
+        #weights = np.random.uniform(1, 3, size=n)
+
+        # Create a DataFrame
+        #data = pd.DataFrame({
+        #    'd': d,
+        #    'y': y,
+        #    'z': z,
+        #    'z2' : z2,
+        #    'c1': c1,
+        #    'c2': c2,
+        #    'cluster': cluster,
+        #   'weights' : weights
+        #})
+
+        #vcov_detail = "iid"
+        #fit_ols = feols("y ~ 1 + d + c1 + c2", data=data, vcov=vcov_detail)
+
+
+        #fit_iv = feols("y ~ 1 + c1  + c2 | d ~ z ", data=data,
+        #         vcov=vcov_detail, weights= "weights")
+        #fit_iv.IV_Diag()
+
+        #print("(Unadjusted) F stat : " ,fit_iv._f_stat_1st_stage)
+        #print("Effective F stat : " ,fit_iv._eff_F )
+        ## The example above generates the following results
+        ## (Unadjusted) F stat :  52.81535560457482
+        ## Effective F stat :  48.661542741328205
+        """
+        ##############################################################
+
+        self.first_stage()
```

Comment:
Should we not call `self.first_stage()` outside of `IV_Diag`? My intuition would have been to call `IV_Diag` within `first_stage()` and not the other way around? 🤔

Reply:
Yes, you are right. I will handle this!